src/OSPSuite.Core/Domain/Formulas/Formula.cs

```diff
@@ -234,20 +233,40 @@ public string FormulaString
       /// </exception>
       public virtual void Validate(string formulaString)
       {
-         var formulaParser = ExplicitFormulaParserCreator(ObjectPaths.Select(path => path.Alias), new string[0]);
+         var formulaParser = ExplicitFormulaParserCreator(UsedVariableNames, new string[0]);
          formulaParser.FormulaString = formulaString ?? string.Empty;
          formulaParser.Parse();
       }
 
       /// <summary>
-      ///    Validates the current formulat string. Throws an
+      ///    Validates the current formula string. Throws an
       /// </summary>
       /// <exception cref="OSPSuiteException"> is thrown if the current FormulaString cannot be successfully parsed </exception>
       public virtual void Validate()
       {
          Validate(FormulaString);
       }
 
-      public static Func<IEnumerable<string>, IEnumerable<string>,  IExplicitFormulaParser> ExplicitFormulaParserCreator { get; set; } = (variableNames, parameterNames) => new NullExplicitFormulaParser();
+      public virtual (bool valid, string validationMessage) IsValid()
+      {
+         return IsValid(FormulaString);
+      }
+
+      public virtual (bool valid, string validationMessage) IsValid(string formulaString)
+      {
+         try
+         {
+            Validate(formulaString);
```

Comment:
This API with the exception is very unnatural. Time to make it more user friendly